helm-frontend/src/components/MiniLeaderboardTables.tsx

```diff
@@ -0,0 +1,303 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useEffect, useState } from "react";
+import { ChevronUpDownIcon } from "@heroicons/react/24/solid";
+import type GroupsTable from "@/types/GroupsTable";
+import RowValue from "@/components/RowValue";
+import Schema from "@/types/Schema";
+import getSchema from "@/services/getSchema";
+
+interface Props {
+  groupsTables: GroupsTable[];
+  activeGroup: number;
+  ignoreHref?: boolean;
+  sortable?: boolean;
+  sortFirstMetric?: boolean;
+  modelsToFilter?: string[];
+  numModelsToAutoFilter?: number;
+  filteredCols?: any[];
+}
+
+export default function MiniLeaderboardTables({
+  groupsTables,
+  activeGroup,
+  ignoreHref = false,
+  sortable = true,
+  sortFirstMetric = true,
+  filteredCols = [],
+  modelsToFilter = [],
+  numModelsToAutoFilter = 0, // if non-zero, sets how many models to filter down to (ranked by first column)
+}: Props) {
+  const [activeSortColumn, setActiveSortColumn] = useState<number | undefined>(
+    sortFirstMetric ? 1 : undefined,
+  );
+  const [activeGroupsTable, setActiveGroupsTable] = useState<GroupsTable>({
+    ...groupsTables[activeGroup],
+  });
+  const [sortDirection, setSortDirection] = useState<number>(1);
+  const [filteredModels, setFilteredModels] =
+    useState<string[]>(modelsToFilter);
```

Comment:
These are pre-existing issues, but overall the design of the props / state is not great.

Props: `groupTable` (i.e. only pass in the active table), `ignoreHref` (perhaps it can be deleted also), `sortColumn`, `filteredCols`, `numModelsToAutoFilter`
State: only `schema`!

There shouldn't be any more state needed - everything should be derived from props! In particular, the user cannot choose the sort order for the mini leaderboard, so the sort column does not need to be state.

I'll approve for now but we should address this later.

Reply:
Agree, can fix in follow-up (removed ignoreHref for now though)